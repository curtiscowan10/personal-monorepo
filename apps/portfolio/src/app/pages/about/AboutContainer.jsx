import React from 'react';
import About from './About';

const AboutContainer = () => {
  const skills = [
    { name: 'React', level: 90 },
    { name: 'JavaScript', level: 95 },
    { name: 'Node.js', level: 85 },
    { name: 'Material-UI', level: 88 },
    { name: 'SCSS/CSS', level: 92 },
    { name: 'Git', level: 90 },
  ];

  const experiences = [
    {
      title: 'Senior Frontend Developer',
      company: 'Tech Company',
      period: '2022 - Present',
      description:
        'Leading frontend development projects using React, TypeScript, and modern web technologies.',
      type: 'work',
    },
    {
      title: 'Full Stack Developer',
      company: 'Digital Agency',
      period: '2020 - 2022',
      description:
        'Developed full-stack web applications using React, Node.js, and various databases.',
      type: 'work',
    },
    {
      title: 'Computer Science Degree',
      company: 'University',
      period: '2016 - 2020',
      description:
        'Bachelor of Science in Computer Science with focus on software engineering and web development.',
      type: 'education',
    },
  ];

  const interests = [
    'Web Development',
    'UI/UX Design',
    'Open Source',
    'Machine Learning',
    'Photography',
    'Travel',
    'Music',
    'Gaming',
  ];

  return (
    <About skills={skills} experiences={experiences} interests={interests} />
  );
};

export default AboutContainer;
