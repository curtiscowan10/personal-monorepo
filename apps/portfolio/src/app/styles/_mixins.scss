// _mixins.scss - Reusable SCSS Mixins

@use 'variables' as *;
@use 'sass:color';
@use 'sass:math';

// Media Query Mixins
@mixin mobile {
  @media (max-width: #{$breakpoint-md - 1px}) {
    @content;
  }
}

@mixin tablet {
  @media (min-width: #{$breakpoint-md}) and (max-width: #{$breakpoint-lg - 1px}) {
    @content;
  }
}

@mixin desktop {
  @media (min-width: #{$breakpoint-lg}) {
    @content;
  }
}

@mixin mobile-only {
  @media (max-width: #{$breakpoint-sm - 1px}) {
    @content;
  }
}

// Layout Mixins
@mixin flex-center {
  display: flex;
  align-items: center;
  justify-content: center;
}

@mixin flex-between {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

@mixin flex-column {
  display: flex;
  flex-direction: column;
}

@mixin flex-column-center {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

// Container Mixins
@mixin container {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 $spacing-sm;

  @include tablet {
    padding: 0 $spacing-lg;
  }

  @include desktop {
    padding: 0 $spacing-xl;
  }
}

@mixin section-spacing {
  padding: $spacing-xl 0;

  @include mobile {
    padding: $spacing-lg 0;
  }
}

// Card Mixins
@mixin card {
  background: $color-white;
  border-radius: $border-radius-md;
  box-shadow: $shadow-base;
  overflow: hidden;
}

@mixin card-hover {
  transition: all $transition-base;

  &:hover {
    box-shadow: $shadow-lg;
    transform: translateY(-2px);
  }
}

@mixin card-content {
  padding: $spacing-lg;

  @include mobile {
    padding: $spacing-md;
  }
}

// Button Mixins
@mixin button-base {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: $spacing-sm $spacing-lg;
  border: none;
  border-radius: $border-radius-md;
  font-weight: $font-weight-medium;
  text-decoration: none;
  cursor: pointer;
  transition: all $transition-base;

  &:disabled {
    opacity: 0.6;
    cursor: not-allowed;
  }
}

@mixin button-primary {
  @include button-base;
  background-color: $color-secondary;
  color: $color-white;

  &:hover:not(:disabled) {
    background-color: color.adjust($color-secondary, $lightness: -10%);
  }
}

@mixin button-secondary {
  @include button-base;
  background-color: transparent;
  color: $color-secondary;
  border: 2px solid $color-secondary;

  &:hover:not(:disabled) {
    background-color: $color-secondary;
    color: $color-white;
  }
}

@mixin button-outline {
  @include button-base;
  background-color: transparent;
  border: 2px solid currentColor;

  &:hover:not(:disabled) {
    background-color: currentColor;
    color: $color-white;
  }
}

// Text Mixins
@mixin text-center {
  text-align: center;
}

@mixin text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

@mixin text-gradient {
  background: linear-gradient(135deg, $color-secondary, $color-accent);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  background-clip: text;
}

// Hero Section Mixins
@mixin hero-section {
  // All CSS declarations first
  background: linear-gradient(135deg, $color-gray-50 0%, $color-gray-100 100%);
  border-radius: $border-radius-md;
  margin-bottom: $spacing-xl;
  text-align: center;

  // Then includes and nested rules
  @include section-spacing;

  @include mobile {
    padding: $spacing-xl 0;
  }
}

@mixin hero-avatar {
  width: 120px;
  height: 120px;
  margin: 0 auto $spacing-md auto;
  background-color: $color-accent;
  font-size: $font-size-3xl;
  border-radius: $border-radius-full;
  @include flex-center;
}

// Form Mixins
@mixin form-input {
  width: 100%;
  padding: $spacing-sm;
  border: 1px solid $color-gray-300;
  border-radius: $border-radius-sm;
  font-size: $font-size-base;
  transition: border-color $transition-base;

  &:focus {
    outline: none;
    border-color: $color-secondary;
    box-shadow: 0 0 0 3px rgba($color-secondary, 0.1);
  }
}

@mixin form-label {
  display: block;
  margin-bottom: $spacing-xs;
  font-weight: $font-weight-medium;
  color: $color-gray-700;
}

// Animation Mixins
@mixin fade-in {
  opacity: 0;
  animation: fadeIn $transition-base forwards;
}

@keyframes fadeIn {
  to {
    opacity: 1;
  }
}

@mixin slide-up {
  transform: translateY(20px);
  opacity: 0;
  animation: slideUp $transition-base forwards;
}

@keyframes slideUp {
  to {
    transform: translateY(0);
    opacity: 1;
  }
}

// Social Button Mixins
@mixin social-button($color) {
  @include button-outline;
  color: $color;
  border-color: $color;

  &:hover:not(:disabled) {
    background-color: $color;
    color: $color-white;
  }
}

// Grid Mixins
@mixin grid($columns: 1, $gap: $spacing-lg) {
  display: grid;
  grid-template-columns: repeat($columns, 1fr);
  gap: $gap;
}

@mixin responsive-grid($mobile: 1, $tablet: 2, $desktop: 3, $gap: $spacing-lg) {
  @include grid($mobile, $gap);

  @include tablet {
    grid-template-columns: repeat($tablet, 1fr);
  }

  @include desktop {
    grid-template-columns: repeat($desktop, 1fr);
  }
}
