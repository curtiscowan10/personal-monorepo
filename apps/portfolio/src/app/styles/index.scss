// index.scss - Main styles entry point

// Import variables and mixins first
@use 'variables' as *;
@use 'mixins' as *;

// Global styles
* {
  box-sizing: border-box;
}

body {
  font-family: Inter, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto,
    sans-serif;
  margin: 0;
  padding: 0;
  line-height: 1.6;
  color: $color-primary;
  background-color: $color-white;
}

// Global utility classes following BEM-like naming
.u-text-center {
  @include text-center;
}

.u-text-truncate {
  @include text-truncate;
}

.u-text-gradient {
  @include text-gradient;
}

.u-fade-in {
  @include fade-in;
}

.u-slide-up {
  @include slide-up;
}

// Global component classes
.c-container {
  @include container;
}

.c-section {
  @include section-spacing;
}

.c-card {
  @include card;

  &--hover {
    @include card-hover;
  }

  &__content {
    @include card-content;
  }
}

.c-button {
  @include button-base;

  &--primary {
    @include button-primary;
  }

  &--secondary {
    @include button-secondary;
  }

  &--outline {
    @include button-outline;
  }
}

.c-form {
  &__input {
    @include form-input;
  }

  &__label {
    @include form-label;
  }
}

.c-grid {
  &--responsive {
    @include responsive-grid;
  }

  &--2-col {
    @include responsive-grid(1, 2, 2);
  }

  &--3-col {
    @include responsive-grid(1, 2, 3);
  }
}
