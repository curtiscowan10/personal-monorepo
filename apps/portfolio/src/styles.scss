/* Main styles entry point - imports the complete styling system */
@use 'app/styles/index';
