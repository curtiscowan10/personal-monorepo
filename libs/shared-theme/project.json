{
  "name": "shared-theme",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/shared-theme/src",
  "projectType": "library",
  "tags": []
}
