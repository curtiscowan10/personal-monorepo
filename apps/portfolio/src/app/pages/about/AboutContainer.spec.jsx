import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import { theme } from '@personal-monorepo/shared-theme';
import AboutContainer from './AboutContainer';

const renderWithProviders = (component) => {
  return render(
    <BrowserRouter>
      <ThemeProvider theme={theme}>{component}</ThemeProvider>
    </BrowserRouter>
  );
};

describe('AboutContainer', () => {
  it('should render successfully', () => {
    renderWithProviders(<AboutContainer />);
    expect(screen.getByTestId('about-page')).toBeTruthy();
  });

  it('should render the About page with header section', () => {
    renderWithProviders(<AboutContainer />);

    expect(screen.getByTestId('about-page')).toBeTruthy();
    expect(screen.getByText(/I'm a passionate developer/)).toBeTruthy();
    expect(screen.getByText('A')).toBeTruthy(); // Avatar
  });

  it('should render bio section with story content', () => {
    renderWithProviders(<AboutContainer />);

    expect(screen.getByText('My Story')).toBeTruthy();
    expect(screen.getByText(/With over 4 years of experience/)).toBeTruthy();
    expect(screen.getByText(/I believe in writing clean/)).toBeTruthy();
    expect(screen.getByText(/When I'm not coding/)).toBeTruthy();
  });

  it('should render technical skills section with all skills', () => {
    renderWithProviders(<AboutContainer />);

    expect(screen.getByText('Technical Skills')).toBeTruthy();

    const expectedSkills = [
      { name: 'React', level: 90 },
      { name: 'JavaScript', level: 95 },
      { name: 'Node.js', level: 85 },
      { name: 'Material-UI', level: 88 },
      { name: 'SCSS/CSS', level: 92 },
      { name: 'Git', level: 90 },
    ];

    expectedSkills.forEach((skill) => {
      expect(screen.getByText(skill.name)).toBeTruthy();
      expect(screen.getByText(`${skill.level}%`)).toBeTruthy();
    });
  });

  it('should render experience and education section', () => {
    renderWithProviders(<AboutContainer />);

    expect(screen.getByText('Experience & Education')).toBeTruthy();

    // Check work experiences
    expect(screen.getByText('Senior Frontend Developer')).toBeTruthy();
    expect(screen.getByText('Tech Company')).toBeTruthy();
    expect(screen.getByText('2022 - Present')).toBeTruthy();

    expect(screen.getByText('Full Stack Developer')).toBeTruthy();
    expect(screen.getByText('Digital Agency')).toBeTruthy();
    expect(screen.getByText('2020 - 2022')).toBeTruthy();

    // Check education
    expect(screen.getByText('Computer Science Degree')).toBeTruthy();
    expect(screen.getByText('University')).toBeTruthy();
    expect(screen.getByText('2016 - 2020')).toBeTruthy();
  });

  it('should render interests and hobbies section', () => {
    renderWithProviders(<AboutContainer />);

    expect(screen.getByText('Interests & Hobbies')).toBeTruthy();

    const expectedInterests = [
      'Web Development',
      'UI/UX Design',
      'Open Source',
      'Machine Learning',
      'Photography',
      'Travel',
      'Music',
      'Gaming',
    ];

    expectedInterests.forEach((interest) => {
      expect(screen.getByText(interest)).toBeTruthy();
    });
  });

  it('should display work and education icons correctly', () => {
    const { container } = renderWithProviders(<AboutContainer />);

    // Should have work icons for work experiences
    expect(
      container.querySelectorAll('.about__experience-card-icon--work')
    ).toHaveLength(2);

    // Should have school icon for education
    expect(
      container.querySelectorAll('.about__experience-card-icon--school')
    ).toHaveLength(1);
  });

  it('should have correct CSS classes applied', () => {
    const { container } = renderWithProviders(<AboutContainer />);

    expect(container.querySelector('.about')).toBeTruthy();
    expect(container.querySelector('.about__header')).toBeTruthy();
    expect(container.querySelector('.about__bio-card')).toBeTruthy();
    expect(container.querySelector('.about__skills-card')).toBeTruthy();
    expect(container.querySelector('.about__experience')).toBeTruthy();
    expect(container.querySelector('.about__interests')).toBeTruthy();
  });

  it('should render progress bars for skills', () => {
    const { container } = renderWithProviders(<AboutContainer />);

    // Should have LinearProgress components for each skill
    const progressBars = container.querySelectorAll(
      '.about__skill-item-progress'
    );
    expect(progressBars.length).toBe(6); // 6 skills
  });

  it('should match snapshot', () => {
    const { container } = renderWithProviders(<AboutContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
