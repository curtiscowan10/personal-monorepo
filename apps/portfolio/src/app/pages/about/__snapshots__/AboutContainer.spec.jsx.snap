// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutContainer > should match snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg about css-5c1adp-MuiContainer-root"
>
  <div
    class="about__header MuiBox-root css-0"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault about__avatar css-1qcqc3p-MuiAvatar-root"
    >
      A
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom custom-title css-jp450z-MuiTypography-root"
    >
      About Me
    </h1>
    <h5
      class="MuiTypography-root MuiTypography-h5 about__subtitle css-rgpxvr-MuiTypography-root"
    >
      I'm a passionate developer who loves creating beautiful, functional web applications that solve real-world problems and provide exceptional user experiences.
    </h5>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-4 css-kqg16c-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root about__bio-card css-r6ekfc-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardContent-root about__bio-card-content css-1lt5qva-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-18p8awu-MuiTypography-root"
          >
            My Story
          </h4>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-dgou5o-MuiTypography-root"
          >
            With over 4 years of experience in web development, I've had the privilege of working on diverse projects ranging from small business websites to large-scale enterprise applications.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-dgou5o-MuiTypography-root"
          >
            I believe in writing clean, maintainable code and staying up-to-date with the latest technologies and best practices. My approach combines technical expertise with creative problem-solving to deliver solutions that not only work well but also provide great user experiences.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1iw893a-MuiTypography-root"
          >
            When I'm not coding, you can find me exploring new technologies, contributing to open-source projects, or enjoying outdoor activities.
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root about__skills-card css-r6ekfc-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardContent-root about__skills-card-content css-1lt5qva-MuiCardContent-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-18p8awu-MuiTypography-root"
          >
            Technical Skills
          </h4>
          <div
            class="about__skills-card-list MuiBox-root css-0"
          >
            <div
              class="about__skill-item MuiBox-root css-0"
            >
              <div
                class="about__skill-item-header MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pwc0nj-MuiTypography-root"
                >
                  React
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-15gs4p4-MuiTypography-root"
                >
                  90
                  %
                </p>
              </div>
              <span
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="90"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate about__skill-item-progress css-1q12wv5-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-26uux8-MuiLinearProgress-bar1"
                  style="transform: translateX(-10%);"
                />
              </span>
            </div>
            <div
              class="about__skill-item MuiBox-root css-0"
            >
              <div
                class="about__skill-item-header MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pwc0nj-MuiTypography-root"
                >
                  JavaScript
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-15gs4p4-MuiTypography-root"
                >
                  95
                  %
                </p>
              </div>
              <span
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="95"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate about__skill-item-progress css-1q12wv5-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-26uux8-MuiLinearProgress-bar1"
                  style="transform: translateX(-5%);"
                />
              </span>
            </div>
            <div
              class="about__skill-item MuiBox-root css-0"
            >
              <div
                class="about__skill-item-header MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pwc0nj-MuiTypography-root"
                >
                  Node.js
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-15gs4p4-MuiTypography-root"
                >
                  85
                  %
                </p>
              </div>
              <span
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="85"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate about__skill-item-progress css-1q12wv5-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-26uux8-MuiLinearProgress-bar1"
                  style="transform: translateX(-15%);"
                />
              </span>
            </div>
            <div
              class="about__skill-item MuiBox-root css-0"
            >
              <div
                class="about__skill-item-header MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pwc0nj-MuiTypography-root"
                >
                  Material-UI
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-15gs4p4-MuiTypography-root"
                >
                  88
                  %
                </p>
              </div>
              <span
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="88"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate about__skill-item-progress css-1q12wv5-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-26uux8-MuiLinearProgress-bar1"
                  style="transform: translateX(-12%);"
                />
              </span>
            </div>
            <div
              class="about__skill-item MuiBox-root css-0"
            >
              <div
                class="about__skill-item-header MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pwc0nj-MuiTypography-root"
                >
                  SCSS/CSS
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-15gs4p4-MuiTypography-root"
                >
                  92
                  %
                </p>
              </div>
              <span
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="92"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate about__skill-item-progress css-1q12wv5-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-26uux8-MuiLinearProgress-bar1"
                  style="transform: translateX(-8%);"
                />
              </span>
            </div>
            <div
              class="about__skill-item MuiBox-root css-0"
            >
              <div
                class="about__skill-item-header MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1pwc0nj-MuiTypography-root"
                >
                  Git
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-15gs4p4-MuiTypography-root"
                >
                  90
                  %
                </p>
              </div>
              <span
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="90"
                class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate about__skill-item-progress css-1q12wv5-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  class="MuiLinearProgress-bar MuiLinearProgress-bar1 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-26uux8-MuiLinearProgress-bar1"
                  style="transform: translateX(-10%);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="about__experience MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom about__experience-title css-hedpeh-MuiTypography-root"
    >
      Experience & Education
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-spacing-xs-3 css-1u8xl8z-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root about__experience-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root about__experience-card-content css-1lt5qva-MuiCardContent-root"
          >
            <div
              class="about__experience-card-icon MuiBox-root css-0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium about__experience-card-icon about__experience-card-icon--work css-1umw9bq-MuiSvgIcon-root"
                data-testid="WorkIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2m-6 0h-4V4h4z"
                />
              </svg>
            </div>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-f6ezis-MuiTypography-root"
            >
              Senior Frontend Developer
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-g470jz-MuiTypography-root"
            >
              Tech Company
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-3w4f7o-MuiTypography-root"
            >
              2022 - Present
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 about__experience-card-description css-1avfp98-MuiTypography-root"
            >
              Leading frontend development projects using React, TypeScript, and modern web technologies.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root about__experience-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root about__experience-card-content css-1lt5qva-MuiCardContent-root"
          >
            <div
              class="about__experience-card-icon MuiBox-root css-0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium about__experience-card-icon about__experience-card-icon--work css-1umw9bq-MuiSvgIcon-root"
                data-testid="WorkIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2m-6 0h-4V4h4z"
                />
              </svg>
            </div>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-f6ezis-MuiTypography-root"
            >
              Full Stack Developer
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-g470jz-MuiTypography-root"
            >
              Digital Agency
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-3w4f7o-MuiTypography-root"
            >
              2020 - 2022
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 about__experience-card-description css-1avfp98-MuiTypography-root"
            >
              Developed full-stack web applications using React, Node.js, and various databases.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root about__experience-card css-r6ekfc-MuiPaper-root-MuiCard-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiCardContent-root about__experience-card-content css-1lt5qva-MuiCardContent-root"
          >
            <div
              class="about__experience-card-icon MuiBox-root css-0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium about__experience-card-icon about__experience-card-icon--school css-1umw9bq-MuiSvgIcon-root"
                data-testid="SchoolIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 13.18v4L12 21l7-3.82v-4L12 17zM12 3 1 9l11 6 9-4.91V17h2V9z"
                />
              </svg>
            </div>
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-f6ezis-MuiTypography-root"
            >
              Computer Science Degree
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-g470jz-MuiTypography-root"
            >
              University
            </h6>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-3w4f7o-MuiTypography-root"
            >
              2016 - 2020
            </p>
            <p
              class="MuiTypography-root MuiTypography-body2 about__experience-card-description css-1avfp98-MuiTypography-root"
            >
              Bachelor of Science in Computer Science with focus on software engineering and web development.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="about__interests MuiBox-root css-0"
  >
    <h2
      class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom about__interests-title css-1a4m65s-MuiTypography-root"
    >
      Interests & Hobbies
    </h2>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root about__interests-card css-r6ekfc-MuiPaper-root-MuiCard-root"
      style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
    >
      <div
        class="MuiCardContent-root about__interests-card-content css-1lt5qva-MuiCardContent-root"
      >
        <div
          class="MuiStack-root about__interests-card-chips css-miphnx-MuiStack-root"
        >
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Web Development
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              UI/UX Design
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Open Source
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Machine Learning
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Photography
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Travel
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Music
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorSecondary MuiChip-outlinedSecondary about__interests-card-chip css-1gh8s9a-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-16cgrcw-MuiChip-label"
            >
              Gaming
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
